OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -456,6 +474,120 @@ void testTrackWithUpdateMarkerWeights()
+    }
+
+    double err = 0.05;
+    SimTK::RowVector_<SimTK::Vec3> biases(3, SimTK::Vec3(0));
+    // bias m0
+    biases[0] += SimTK::Vec3(0, err, 0);
+    cout << "biases: " << biases << endl;
+
+    auto markerTable = generateMarkerDataFromModelAndStates(*pendulum,
+                        states,
+                        biases,
+                        0.0,
+                        true);
+
+    SimTK::Vector_<SimTK::Vec3> unusedCol(N, SimTK::Vec3(0.987654321));
+
+    auto usedMarkerNames = markerTable.getColumnLabels();
+
+    // add an unused marker to the marker data
+    markerTable.appendColumn("unused", unusedCol);
+
+    cout << "Before:\n" << markerTable << endl;
+    
+    // re-order "observed" marker data
+    SimTK::Matrix_<SimTK::Vec3> dataGutsCopy = markerTable.getMatrix();
+    int last = dataGutsCopy.ncol() - 1;
+    // swap first and last columns 
+    markerTable.updMatrix()(0) = dataGutsCopy(last);
+    markerTable.updMatrix()(last) = dataGutsCopy(0);
+    auto columnNames = markerTable.getColumnLabels();
+    markerTable.setColumnLabel(0, columnNames[last]);
+    markerTable.setColumnLabel(last, columnNames[0]);
+    columnNames = markerTable.getColumnLabels();
+
+    // Inject NaN in "observations" of "mR" marker data
+    for (int i = 4; i < 7; ++i) {
+        markerTable.updMatrix()(i, 1) = SimTK::NaN;
+    }
+
+    cout << "After reorder and NaN injections:\n" << markerTable << endl;
+
+    MarkersReference markersRef(markerTable);
+    int nmr = markersRef.getNumRefs();
+    auto& markerNames = markersRef.getNames();
+    cout << markerNames << endl;
+
+    SimTK::Array_<CoordinateReference> coordRefs;
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    InverseKinematicsSolver ikSolver(*pendulum, markersRef, coordRefs);
+    double tol = 1e-4;
+    ikSolver.setAccuracy(tol);
+    ikSolver.assemble(state);
+
+    int nm = ikSolver.getNumMarkersInUse();
+
+    SimTK::Array_<double> markerErrors(nm);
+    for (unsigned i = 0; i < markersRef.getNumFrames(); ++i) {
+        state.updTime() = i*dt;
+        ikSolver.track(state);
+
+        //get the marker errors
+        ikSolver.computeCurrentMarkerErrors(markerErrors);
+
+        int nme = markerErrors.size();
+
+        SimTK_ASSERT_ALWAYS(nme = nm,
+            "InverseKinematicsSolver failed to account "
+            "for unused marker reference (observation).");
+
+        cout << "time: " << state.getTime() << " |";
+        auto namesIter = usedMarkerNames.begin();
+        for (int j = 0; j < nme; ++j) {
+            const auto& markerName = ikSolver.getMarkerNameForIndex(j);
+            cout << " " << markerName << " error = " << markerErrors[j];
+
+            SimTK_ASSERT_ALWAYS( *namesIter++ != "unused",
+                "InverseKinematicsSolver failed to ignore "
+                "unused marker reference (observation).");
+            
+            if (markerName == "m0") {// marker we biased shoulder the error
```

Comment:
Could you rephrase "marker we biased shoulder the error"? I don't understand what this means.